Add input to skip labeling already labeled issues

brcrista opened the pull request:
Where we use this, we have the action add a "needs triage" label to all new issues.  Often, we already know how an issue should be triaged when we open it, so the labeling by the bot is redundant and just has to be removed.

This is adding an input similar to `ignore-if-assigned` to ignore if an issued is labeled.

**Testing**
https://github.com/brcrista/labeler-test/issues

brcrista committed:
Add syntax highlighting in README

brcrista committed:
ignore-add-if-labeled

brcrista committed:
Logic? what's that

brcrista committed:
Empty list is not falsy :/

brcrista committed:
fix log message at the end

brcrista commented on action.yml:
```diff
@@ -11,8 +11,11 @@ inputs:
-    description: "True/False value to indicate if no labels should be added or removed if there is an asignee to the issue"
+    description: "True/False value to indicate if no labels should be added or removed if there is an assignee to the issue."
     required: true
+  ignore-add-if-labeled:
```
Open to naming / semantics here.  I thought `ignore-if-labeled` looked good because it parallels `ignore-if-assigned`, but not sure if it makes sense to ignore removing labels 🤔

andymckay replied:
Naming arrgh! I feel like `ignore-if-labeled` works and is shorter and parallels nicely, so I'd go with that.

brcrista replied:
After sleeping on it, I arrived at the same conclusion.  People can use multiple actions / workflows for deeper logic.

andymckay commented on label.js:
```diff
@@ -30,12 +31,15 @@ async function label() {
   if (ignoreIfAssigned) {
     // check if the issue has been assigned to anyone
-    if (updatedIssueInformation.data.assignees.length != 0) {
+    if (updatedIssueInformation.data.assignees.length !== 0) {
```
Ha good one.

andymckay commented on label.js:
```diff
@@ -30,12 +31,15 @@ async function label() {
 
   let labels = updatedIssueInformation.data.labels.map(label => label.name);
+  if (ignoreAddIfLabeled && labels.length > 0) {
+    labelsToAdd = [];
```
This works, I was thinking that we could move the for loop up into here and its all double negatives. So 👍

brcrista replied:
Yeah, this also helps make sure the log line at the end prints the correct info.

andymckay reviewed (approved):
Lets go for `ignore-if-labeled`. Otherwise looks great, thanks!

brcrista committed:
rename to `ignore-if-labeled` and change semantics accordingly

brcrista committed:
const

brcrista committed:
Merge branch 'master' into brcrista/ignore-if-labeled

brcrista committed:
Fix duplicated line after merge resolution

brcrista commented:
@andymckay I changed it to `ignore-if-labeled` and updated the semantics accordingly.  Latest test here: https://github.com/brcrista/labeler-test/issues
